software/glasgow/device/hardware.py

```diff
@@ -113,6 +112,19 @@ def __init__(self, serial=None, firmware_filename=None, *, _factory_rev=None):
                 # Give every device we loaded firmware onto a bit of time to reenumerate.
                 time.sleep(1.0)
 
+        return handles
+
+    @classmethod
+    def get_serial_list(cls, firmware_filename=None, *, _factory_rev=None):
+        usb_context = usb1.USBContext()
+        handles = cls._enumerate_devices(usb_context, firmware_filename, _factory_rev)
+        usb_context.close()
+        return handles.keys()
```

Comment:
This returns an iterable of a dict, I'd suggest converting it to a list first.
```suggestion
        return list(handles.keys())
```